arns/arns-token-1.md

```diff
@@ -261,6 +272,142 @@ Send({
+
+##### Parameters
+
+| Name   | Type   | Description                 |
+| ------ | ------ | --------------------------- |
+| Description | string | The new Description for the ANT. |
+
+##### Rules
+
+- Must be an authorized process `Owner` or `Controller`.
+- Must specify a valid `Description` parameter (string) as a message tag.
+- `Description` must not be longer than 512 characters.
+- Should add `X-`forwarded tags to the response notice.
+
+##### Action
+
+```
+Send({
+  Target = "{Process Identifier}",
+  Action = "Set-Description",
+  Description = "ArDrive is an app that makes it easy to upload, download and share your public or private files on Arweave"
+})
+```
+
+##### Responses
+
+**Permission error, not authorized**
+
+```
+{
+  Target = msg.From,
+  Action = "Invalid-Set-Description-Notice",
+  Data = permissionErr,
+  Error = "Set-Description-Error",
+  ["Message-Id"] = msg.Id
+}
+```
+
+**Invalid parameters**
+
+```
+{
+  Target = msg.From,
+  Action = "Invalid-Set-Description-Notice",
+  Data = descriptionRes,
+  Error = "Set-Description-Error",
+  ["Message-Id"] = msg.Id,
+}
+```
+
+**Valid parameters**
+
+```
+{
+  Target = msg.From,
+  Action = "Set-Description-Notice",
+  Data = json.encode({ Description = Description }),
+  ... other forwarded tag name and value pairs
+}
+```
+
+#### Set-Keywords
+
+Updates the `Keywords` of this ANT.
+
+Executable by the process `Owner` or an authorized user in the `Controllers` table.
+
+##### Parameters
+
+| Name   | Type   | Description                 |
+| ------ | ------ | --------------------------- |
+| Keywords | table | The new Keywords for the ANT. |
+
+##### Rules
+
+- Must be an authorized process `Owner` or `Controller`.
+- Must specify a valid `Keywords` parameter (table) as a message tag.
+- Each `Keyword` must not be longer than 32 characters.
+- Each `Keyword` must consiste of alphanumeric characters, dashes or underscores.
+- Each `Keyword` must not include spaces.
+- Each `Keyword` must be unique in the table of `Keywords`.
+- There must not be more than 16 total `Keywords`.
+- Should add `X-`forwarded tags to the response notice.
+
+##### Action
+
+```
+Send({
+  Target = "{Process Identifier}",
+  Action = "Set-Description",
+  Description = "ArDrive is an app that makes it easy to upload, download and share your public or private files on Arweave"
+})
+```
+
+##### Responses
+
+**Permission error, not authorized**
+
+```
+{
+  Target = msg.From,
+  Action = "Invalid-Set-Description-Notice",
+  Data = permissionErr,
+  Error = "Set-Description-Error",
+  ["Message-Id"] = msg.Id
+}
+```
+
+**Invalid parameters**
+
+```
+{
+  Target = msg.From,
+  Action = "Invalid-Set-Description-Notice",
+  Data = descriptionRes,
+  Error = "Set-Description-Error",
+  ["Message-Id"] = msg.Id,
+}
+```
+
+**Valid parameters**
+
+```
+{
+  Target = msg.From,
+  Action = "Set-Description-Notice",
+  Data = json.encode({ Description = Description }),
```

Comment:
In our other apis when requesting a string we return a string - recommend the same here.

Reply:
What do you mean?  I'm just following the same pattern for Set-Name and Set-Ticker.  Unless I misinterpretted it.

Reply:
Sorry, that is correct, memory was failing me.
Small nit though since I noticed it doesnt match up:
```suggestion
  Data = json.encode({ description = Description }),
```